components/elm/docs/dev-guide/testing.md

```diff
@@ -0,0 +1,183 @@
+
+For the purpose of these document, let's assume the developer branch is `bishtgautam/lnd/emi-example`.
+
+## Step-1: Generate Baselines
+
+Find the starting hash on `master` by switching to the developer branch and looking at the Git
+graph.
+
+```
+cd <e3sm-dir>
+
+# Switch to your branch
+git checkout bishtgautam/lnd/emi-example
+
+# Let's look at the graph
+git log --oneline --decorate --graph
+* 9a2cd8459 (HEAD -> bishtgautam/lnd/emi-example, origin/bishtgautam/lnd/emi-example) Updates the Stub EM code
+* d11fb8c52 Adds a readme and makefile for EMI Demo code
+* 15a0e4b8d Minor update to the EMI demo code
+* 63e2f6dcd Minor fix to ELM configure script
+* 551fc79cc Updates ELM stub and demo code to use CNCarbonStateType
+* 8c9d41f87 Adds code to exchange CNCarbonStateType
+* 0000ded51 Minor fix
+* 2baa4ebec Adds fixes to cmake
+* f77ae2cea Rename few cmake files
+* b48b66dd6 Adds a demo for a Stub EM
+* fa94aea0d Fixes length of variable to store name of EMI data
+* cca0e0be1 Adds capability to print all EMI Data in a list
+* beb1fe9ee Adds .gitignore files for EMI
+* 301acfe7a Renames ExternalModelInterfaceDataMod.F90
+* f0eb5b40b Renames ExternalModelInterfaceDataDimensionMod.F90
+* 076100882 Restructures ELM's EMI directory
+*   cfb7fc2b2 Merge branch 'qzhu-lbl/lnd/ch4_inundation_bugfix' (PR #2814)
+|\
+| * ec99b1cb4 (origin/qzhu-lbl/lnd/ch4_inundation_bugfix)  - fix ch4 inundatation parameter read in bug
+* |   c59c98116 Merge branch 'darincomeau/mpaso/enable_eddystats' (PR #2821)
+|\ \
+| * | c21bb46a4 Turn on eddy stats for oEC60to30v3 and oEC60to30v3wLI ocn grids by default
+* | |   a2dd2fce5 Merge pull request #2811 from E3SM-Project/jayeshkrishna/pio2_cime_changes
+```
+
+The above Git graph tells us that `bishtgautam/lnd/emi-example` started with the `cfb7fc2b2` hash on the `master`. So, we will generate the baselines using `cfb7fc2b2`. Checkout `cfb7fc2b2` and update the submodels via:
+
+```
+git checkout cfb7fc2b2
+Note: checking out 'cfb7fc2b2'.
+
+You are in 'detached HEAD' state. You can look around, make experimental
+changes and commit them, and you can discard any commits you make in this
+state without impacting any branches by performing another checkout.
+
+If you want to create a new branch to retain commits you create, you may
+do so (now or later) by using -b with the checkout command again. Example:
+
+  git checkout -b <new-branch-name>
+
+HEAD is now at cfb7fc2b2... Merge branch 'qzhu-lbl/lnd/ch4_inundation_bugfix' (PR #2814)
+
+git submodule update --init
+```
+
+Now we will use `cime/scripts/create_test` to generate baseline for the `e3sm_land_developer`
+test suite.
+
+```
+cd cime/scripts
+```
+
+A few things that you need to decide at this stage include the directory location where the baselines will be saved, the name and ID for the baselines, the project allocation that you will use for running the simulation, if you would like to recieve email notirications about tests, etc.
```

Comment:
Couple of typos:   
"recieve" should be "receive" 
"notirications" -> "notifications".